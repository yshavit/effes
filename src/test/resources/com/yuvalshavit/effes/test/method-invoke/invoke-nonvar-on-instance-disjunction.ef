type ATrue:
  mapped -> True: return True

type AFalse:
  mapped -> False: return False

type AWrap(value: ATrue | AFalse)

unwrap (b: AWrap) -> True | False:
  return b.value.mapped

main:
  print (unwrap AWrap(ATrue))
  print (unwrap AWrap(AFalse))
