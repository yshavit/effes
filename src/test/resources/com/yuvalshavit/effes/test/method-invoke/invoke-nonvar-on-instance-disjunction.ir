def unwrap (b: AWrap) -> False | True:
  ReturnStatement
    CaseExpression
      case:
        AssignExpression
          assignTo: $0 (pos=0)
          InstanceArg
            target:
              VarExpression
                arg: b
                pos: 0
            arg: [0] value: AFalse | ATrue
      of AFalse:
        MethodInvoke
          method: AFalse::mapped
          arg0:
            CastExpression
              castTo: AFalse
              VarExpression
                var: $0
                pos: 0
      of ATrue:
        MethodInvoke
          method: ATrue::mapped
          arg0:
            CastExpression
              castTo: ATrue
              VarExpression
                var: $0
                pos: 0
def AFalse::mapped () -> False:
  ReturnStatement
    CtorInvoke
      type: False
def ATrue::mapped () -> True:
  ReturnStatement
    CtorInvoke
      type: True
