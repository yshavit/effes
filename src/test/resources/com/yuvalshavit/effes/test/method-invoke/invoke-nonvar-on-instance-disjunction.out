Your argument was: True
Your argument was: False
>> null
